monkey/monkey_island/cc/services/authentication_service/otp/otp_generator.py

```diff
@@ -0,0 +1,34 @@
+import string
+from datetime import datetime, timedelta, timezone
+
+from common.utils.code_utils import secure_generate_random_string
+from monkey_island.cc.models import OTP
+from monkey_island.cc.repositories import IOTPRepository
+
+OTP_EXPIRATION_TIME = 2 * 60
+
+
+class OTPGenerator:
+    """
+    Generates OTPs
+    """
+
+    def __init__(self, otp_repository: IOTPRepository):
+        self._otp_repository = otp_repository
+
+    def generate_otp(self) -> OTP:
+        """
+        Generates a new OTP
+
+        The generated OTP is saved to the `IOTPRepository`
```

Comment:
This may violate SRP. It seems like maybe the `AuthenticationFacade` can use `OTPGenerator` and `OTPRepository` and provide `AuthenticationFacade.get_new_otp()` (or similar) that wraps the action of "create and store".